File: DSA_in_Python/recursive_binary_search.py

def recursive_binary_search(list, target):
    
    if len(list) == 0:
        return False
    
    else:
        mid_point = len(list) // 2 
        if list[mid_point] == target:
            return True
            
        # return  recursive_binary_search(list[:mid_point-1], target)  if list[mid_point] > target else recursive_binary_search(list[mid_point+1:], target)
        # return (recursive_binary_search(list[:mid_point-1], target), recursive_binary_search(list[mid_point+1:], target) ) [list[mid_point] < target]
        return {True: recursive_binary_search(list[:mid_point-1], target), False:recursive_binary_search(list[mid_point+1:], target) } [list[mid_point] > target]
        # else: 
        #     if list[mid_point] > target: 
        #         return recursive_binary_search(list[:mid_point-1], target)
        #     elif list[mid_point] < target: 
        #         return recursive_binary_search(list[mid_point+1:], target)

numbers = [1,2,3,4,5,6,7,8,9,10]

def verify (result):
    if result is True: 
        print('Target found : ', result)

result = recursive_binary_search(numbers, 3)
verify(result)

File: DSA_in_PHP/recursive_binary_search.php
<?php 

/**
 * Search for an item in an array using recursive binary search algorithm
 * 
 * @param array $list 
 * @param any $target 
 * @return integer||null
 */
 

function recursiveBinarySearch(Array $list , $target)
{
    $list_size = count($list);
   
    if ($list_size == 0)
        return false;
   
   if($list_size==1)
        return $list[0] == $target;
   
   $mid_point = floor ($list_size / 2);

   if($list[$mid_point] == $target )
        return true;

    return $list[$mid_point] > $target ? recursiveBinarySearch(array_slice($list, 0, $mid_point), $target) 
    :  recursiveBinarySearch(array_slice($list, $mid_point), $target);
}

function verify($result){
    echo $result?  "Target found!!": 'Target not found';
}

$numbers = [1,2,3,4,5,6,7,8,9,10];
$result = recursiveBinarySearch($numbers, 10);
verify($result);